src/file-events/cpp/win_fsnotifier.cpp

```diff
@@ -61,9 +61,8 @@ bool WatchPoint::cancel() {
 
 WatchPoint::~WatchPoint() {
     try {
-        if (cancel()) {
-            SleepEx(0, true);
-        }
+        cancel();
```

Comment:
I wonder if the should try cancelling again here when an earlier cancel failed, i.e. not always  set the status to `FINISHED` in `cancel`. WDYT?

Reply:
We could also do that by not observing the status in `cancel()`... That said, let's see if this fixes the errors, and find an explanation of why it does if it does first.